detox/local-cli/templates/jest.js

```diff
@@ -12,32 +13,21 @@ const config = require('../package.json').detox;
 const adapter = require('detox/runners/jest/adapter');
 const specReporter = require('detox/runners/jest/specReporter');
 
-// Set the default timeout
-jest.setTimeout(120000);
-
-jasmine.getEnv().addReporter(adapter);
+detoxCircus.getEnv().addEventsListener(adapter);
 
 // This takes care of generating status logs on a per-spec basis. By default, jest only reports at file-level.
 // This is strictly optional.
-jasmine.getEnv().addReporter(specReporter);
+detoxCircus.getEnv().addEventsListener(specReporter);
+
+// Set the default timeout
+jest.setTimeout(90000);
 
 beforeAll(async () => {
   await detox.init(config);
 }, 300000);
 
 beforeEach(async () => {
-  try {
-    await adapter.beforeEach();
-  } catch (err) {
-    // Workaround for the 'jest-jasmine' runner (default one): if 'beforeAll' hook above fails with a timeout,
-    // unfortunately, 'jest' might continue running other hooks and test suites. To prevent that behavior,
-    // adapter.beforeEach() will throw if detox.init() is still running; that allows us to run detox.cleanup()
-    // in that emergency case and disable calling 'device', 'element', 'expect', 'by' and other Detox globals.
-    // If you switch to 'jest-circus' runner, you can omit this try-catch workaround at all.
-
-    await detox.cleanup();
-    throw err;
-  }
+  await adapter.beforeEach();
```

Comment:
This is awesome!!!

Reply:
Also, if this is the case - perhaps we should consider simplifying the API for getting the `env`, or whichever proxy-object we decide proper here in order to add a listener? (the `detoxCircus.getEnv()` shape of an API was just a means to stay coherent with jasmine's `jasmine.getEnv()`...)

Reply:
Nice idea, let's discuss that soon.

Reply:
Done.